### PR TITLE
fix: define imported package's local name by using flags

### LINKED ISSUE #646: Does `-imports` work? Package alias is inconsistent, and -imports does not fix it.
**Actual behavior** A clear and concise description of what the bug is.

The package alias produced by mockgen is variable for the same inputs. Specifically, sometimes we see
	`extract "github.com/codeclysm/extract/v3"`
and others we see
	`v3 "github.com/codeclysm/extract/v3"`
Also, using `-imports extract=github.com/codeclysm/extract/v3` does not have any effect on the output.

**Expected behavior** A clear and concise description of what you expected to happen.

I expect the same inputs to product the same outputs.
I expect -imports to determine the package alias.

**To Reproduce** Steps to reproduce the behavior

1.  This program creates a mock file with no mention of cheesecake, as expected. Am I using `-imports` correctly, and is that the effect it's supposed to have?

```
package main

import (
	"context"
	"io"

	"github.com/codeclysm/extract/v3"
)

//go:generate go run github.com/golang/mock/mockgen -destination=mock_if.go -package main -source ./gomocktest.go -imports cheesecake=github.com/codeclysm/extract/v3

type IF interface {
	ExtractGz(ctx context.Context, body io.Reader, location string, renamer extract.Renamer) error
	ExtractZip(ctx context.Context, body io.Reader, location string, renamer extract.Renamer) error
}

func main() {
	print("hello world")
}
```

**Additional Information**

-   gomock mode (reflect or source): Source, run from `go generate` although running directly has the same output.
-   gomock version or git ref: github.com/golang/mock v1.6.0
-   golang version: 1.17

**Triage Notes for the Maintainers**

### PR DESCRIPTION
**Pull request by ErfanMomeniii**
fixes [#646](https://github.com/golang/mock/issues/646),
Hi, my friends.
Merging this pull request can fix old bugs related to defining the local names of imported packages by using the -imports flag.
I have also added a test for that, and it works properly.
If you have any ideas or concerns regarding the PR, feel free to tell me.

### CODE REVIEW

**Comment by OscarVanL** [comment]
I ran into this bug today for the first time (in the old gomock/mockgen package). It's frustrating the import names get broken in the generated mocks when using import aliases or dot imports.

File: mockgen/internal/tests/defined_import_local_name/input.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package defined_import_local_name
+
+import (
+	"bytes"
+	"context"
+)
+
+//go:generate mockgen -package defined_import_local_name -destination mock.go -source input.go -imports b_mock=bytes,c_mock=context
+
+type WithDotImports interface {
```
Comment (r-hang) [inline mockgen/internal/tests/defined_import_local_name/input.go:10]: Could we add an additional package to test for stdlib and external imports?

Suggestion: Update the method names to reflect which scenario being tested.

Comment (r-hang) [inline mockgen/internal/tests/defined_import_local_name/input.go:10]: What is DotImports referring to? In the spec I see three cases
```
Import declaration          Local name of Sin

import   "lib/math"         math.Sin
import m "lib/math"         m.Sin
import . "lib/math"         Sin
```

As a reviewer, the DotImport name makes me think this is that last case in the list, but this test seems to be testing named imports (case 2).

**Comment by ErfanMomeniii** [comment]
Dear @r-lang, you are correct. I have renamed it to `withImports` as these tests are not intended for testing the dot imports feature. Thank you.

**Review by sywhang — APPROVED** [review]

**Pull request merged**